README.md

```diff
@@ -1114,6 +1166,56 @@ func main() {
+
+A common point of confusion is that the command string behaves the same as the shell; it does not. In the shell, when you run `foo | bar` or `foo > bar`, that is actually running as a subprocess of your shell (bash, zsh, csh, etc.). When Consul Template spawns the exec process, it runs outside of your shell. This behavior is _different_ from when Consul Template executes the template-specific reload command. If you want the ability to pipe or redirect in the exec command, you will need to spawn the process in subshell, for example:
+
+```javascript
+exec {
+  command = "$SHELL -c 'my-server > /var/log/my-server.log'"
+}
+```
+
+Note that when spawning like this, most shells do not proxy signals to their child by default, so your child process will not receive the signals that Consul Template sends to the shell. You can avoid this by writing a tiny shell wrapper and executing that instead:
+
+```bash
+#!/usr/bin/env bash
+trap "kill -TERM $child" SIGTERM
+
+/bin/my-server -config /tmp/server.conf
+child=$!
+wait "$child"
+```
+
+Alternatively, you can use your shell's exec function directly, if it exists:
+
+```bash
+#!/usr/bin/env bash
+exec /bin/my-server -config /tmp/server.conf > /var/log/my-server.log
+```
+
+There are some additional caveats with Exec Mode, which should be considered carefully before use:
+
+- If the child process dies, the Consul Template process will also die. Consul Template **does not supervise the process!** This is generally the responsibility of the scheduler or init system.
+- The child process must remain in the foreground. This is a requirement for Consul Template to manage the process and send signals.
+- The exec command will only start after _all_ templates have been rendered at least once. One may have multiple templates for a single Consul Template process, all of which must be rendered before the process starts. Consider something like an nginx or apache configuration where both the process configuration file and individual site configuration must be written in order for the service to successfully start.
+- After the child process is started, any change to any dependent template will cause the reload signal to be sent to the child process. This reload signal defaults to SIGHUP, but can be customizing via the CLI or configuration file.
+- When Consul Template is stopped gracefully, it will send the configurable kill signal to the child process. The default value is SIGTERM, but it can be customized via the CLI or configuration file.
+- It is not possible to have more than one exec command (although each template can still have its own reload command).
```

Comment:
so using one CT instance with multiple `template {}` instances would need to move to one CT instance per `exec` 

it would be kinda nice to be able to have multiple `exec` per instance, and then attach an `exec` to a `template` a bit like puppet `notify`/`listen` behavior